stress/shell/src/main/java/alluxio/stress/cli/report/GenerateReport.java

```diff
@@ -0,0 +1,134 @@
+import alluxio.stress.JsonSerializable;
+import alluxio.stress.Summary;
+import alluxio.stress.graph.Graph;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParametersDelegate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Generates a report from summaries.
+ */
+public class GenerateReport {
+  private static final Logger LOG = LoggerFactory.getLogger(GenerateReport.class);
+
+  @ParametersDelegate
+  private Parameters mParameters = new Parameters();
+
+  @com.beust.jcommander.Parameters(separators = "=")
+  private static class Parameters {
+    @Parameter(names = "--input",
+        description = "The input json files of the results. Can be repeated", required = true)
+    private List<String> mInputs;
+
+    @Parameter(names = "--output-dir", description = "The output directory", required = true)
+    private String mOutputDir;
+  }
+
+  /**
+   * @param args the command-line args
+   */
+  public static void main(String[] args) {
+    new GenerateReport().run(args);
+  }
+
+  GenerateReport() {
+  }
+
+  private void run(String[] args) {
+    JCommander jc = new JCommander(this);
+    jc.setProgramName(this.getClass().getSimpleName());
+    try {
+      jc.parse(args);
+    } catch (Exception e) {
+      jc.usage();
+      throw e;
+    }
+
+    List<Summary> inputs =
+        mParameters.mInputs.stream().map(f -> {
+          try {
+            return readJson(f);
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        }).collect(Collectors.toList());
+
+    String className = null;
+    for (Summary summary : inputs) {
+      if (className == null) {
+        className = summary.getClassName();
+      }
+      if (!className.equals(summary.getClassName())) {
+        throw new RuntimeException(
+            "Mismatched input result types: " + className + " , " + summary.getClassName());
+      }
+    }
+
+    File outputDir = new File(mParameters.mOutputDir);
+    outputDir.mkdirs();
+
+    GraphGenerator graphGenerator = inputs.get(0).graphGenerator();
+
+    List<Graph> graphs = graphGenerator.generate(inputs);
+
+    try (PrintWriter writer = new PrintWriter(
+        new FileWriter(Paths.get(outputDir.getAbsolutePath(), "index.html").toString()))) {
+
+      writer.println("<!DOCTYPE html>");
+      writer.println("<head>");
+      writer.println("<script src=\"https://cdn.jsdelivr.net/npm/vega@5\"></script>");
+      writer.println("<script src=\"https://cdn.jsdelivr.net/npm/vega-lite@4\"></script>");
+      writer.println("<script src=\"https://cdn.jsdelivr.net/npm/vega-embed@6\"></script>");
+      writer.println("</head>");
+      writer.println("<body>");
```

Comment:
This seems a bit hacky...hmm. I wonder if there's a better way to write this out rather than putting HTML in java strings within the code.

Maybe we can use a template in the `resources` directory instead and format that?

Reply:
Or use a real HTML parsing/object creation library instead?

Reply:
I had considered a library, but those seemed more heavyweight for a simple file that I was generating. If I wanted more to the generated file, then maybe we need a library or template. I will look into it some...